### PR TITLE
fix: skip setting SdnRemoteArpMacAddress when hns is not enabled

### PR DESCRIPTION
**Pull request by ZetaoZhuang**
**Reason for Change**:

fix: skip setting SdnRemoteArpMacAddress when hns is not enabled
**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [x] adds unit tests
- [ ] relevant PR labels added

**Notes**:

### CODE REVIEW

File: platform/os_windows_test.go [hunk 1]
```diff
@@ -98,3 +99,38 @@ func TestExecuteCommandError(t *testing.T) {
...
+		if strings.Contains(cmd, "Test-Path") {
+			return "true", nil
+		}
+		return "", nil
+	})
+	err := SetSdnRemoteArpMacAddress(mockExecClient)
+	assert.NoError(t, err)
+	assert.Equal(t, true, sdnRemoteArpMacAddressSet)
+	// reset sdnRemoteArpMacAddressSet
+	sdnRemoteArpMacAddressSet = false
```
Comment (ashvindeodhar) [inline platform/os_windows_test.go:135]: why is this needed?

**Review by ashvindeodhar — CHANGES_REQUESTED** [review]

File: platform/os_windows_test.go (hunk 1 above)
Comment (ZetaoZhuang) [reply platform/os_windows_test.go:135]: its not necessary in this single test. but it is required if we want to add more tests for this sdnRemoteArpMacAddressSet

File: cns/service/main.go [hunk 2]
```diff
@@ -724,8 +724,9 @@ func main() {
 		}
 	}
 
-	// Setting the remote ARP MAC address to 12-34-56-78-9a-bc on windows for external traffic
-	err = platform.SetSdnRemoteArpMacAddress()
+	// Setting the remote ARP MAC address to 12-34-56-78-9a-bc on windows for external traffic if HNS is enabled
+	execClient := platform.NewExecClient(nil)
+	err = platform.SetSdnRemoteArpMacAddress(execClient)
```
[the remark this answers is not in the document]
Comment (ZetaoZhuang) [reply cns/service/main.go:729]: passing the execClient to this func so we can pass the mockClient to this func in UT to mock different behaviors

**Review by ashvindeodhar — APPROVED** [review]

File: platform/os_windows.go [hunk 3]
```diff
@@ -190,24 +194,33 @@ func (p *execClient) ExecutePowershellCommand(command string) (string, error) {
 	return strings.TrimSpace(stdout.String()), nil
 }
 
-// SetSdnRemoteArpMacAddress sets the regkey for SDNRemoteArpMacAddress needed for multitenancy
-func SetSdnRemoteArpMacAddress() error {
-	p := NewExecClient(nil)
+// SetSdnRemoteArpMacAddress sets the regkey for SDNRemoteArpMacAddress needed for multitenancy if hns is enabled
+func SetSdnRemoteArpMacAddress(execClient ExecClient) error {
```
Comment (msvik) [inline platform/os_windows.go:198]: Just curious to know why did we change the method signature and pass in ExecClient as parameter?

Comment (ZetaoZhuang) [reply platform/os_windows.go:198]: passing the execClient to this func so we can pass the mockClient to this func in UT to mock different behaviors

**Review by msvik — APPROVED** [review]
looks good overall
